build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -2141,4 +2141,18 @@ message RequestMetadata {
   // There is no expectation that this value will have any particular structure,
   // or equality across invocations, though some client tools may offer these guarantees.
   string configuration_id = 7;
+
+  // The details of other tools involved in handling the request, eg. client-side proxies.
+  // This field MUST NOT include the details of the tool which initially invoked the gRPC
+  // request (ie. it MUST NOT duplicate `RequestMetadata.tool_details`). The order of
+  // tool information in this field has no guarantees or meaning.
+  //
+  // In the case of a request which goes through the following steps:
+  //
+  //    client -> proxy1 -> proxy2 -> server
+  //
+  // `RequestMetadata.tool_details` SHOULD contain a ToolDetails message about `client`,
+  // and `RequestMetadata.additional_handlers` MAY contain two ToolDetails messages,
```

Comment:
I think that both tool_details and additional_handlers probably both SHOULD (not MAY) contain the required information. Technically all of this is optional, in an optional header, but within that scope I don't see a good rationale for making one SHOULD and the other MAY.